src/proofs/weighted_inner_product.rs

```diff
@@ -0,0 +1,915 @@
+#![allow(non_snake_case)]
+
+/*
+
+Copyright 2018 by Kzen Networks
+
+This file is part of bulletproof library
+(https://github.com/KZen-networks/bulletproof)
+
+bulletproof is free software: you can redistribute
+it and/or modify it under the terms of the GNU General Public
+License as published by the Free Software Foundation, either
+version 3 of the License, or (at your option) any later version.
+
+@license GPL-3.0+ <https://github.com/KZen-networks/bulletproof/blob/master/LICENSE>
+*/
+
+// based on the paper: https://eprint.iacr.org/2017/1066.pdf
+use curv::arithmetic::traits::Modulo;
+use curv::cryptographic_primitives::hashing::hash_sha256::HSha256;
+use curv::cryptographic_primitives::hashing::traits::*;
+use curv::elliptic::curves::traits::*;
+use curv::BigInt;
+use curv::{FE, GE};
+
+use Errors::{self, WeightedInnerProdError};
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct WeightedInnerProdArg {
+    L: Vec<GE>,
+    R: Vec<GE>,
+    a_tag: GE,
+    b_tag: GE,
+    r_prime: BigInt,
+    s_prime: BigInt,
+    delta_prime: BigInt,
+}
+
+impl WeightedInnerProdArg {
+    pub fn prove(
+        G: &[GE],
+        H: &[GE],
+        g: &GE,
+        h: &GE,
+        P: &GE,
+        a: &[BigInt],
+        b: &[BigInt],
+        alpha: &BigInt,
+        mut L_vec: Vec<GE>,
+        mut R_vec: Vec<GE>,
+    ) -> WeightedInnerProdArg {
+        let n = G.len();
+        let order = FE::q();
+
+        // All of the input vectors must have the same length.
+        assert_eq!(G.len(), n);
```

Comment:
n is equal to G.len() by definition (line 52)

Reply:
Removed all checks of length of `G`.